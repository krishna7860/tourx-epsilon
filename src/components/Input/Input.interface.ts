import { CSSProperties } from "styled-components";

export interface InputProps {
  value?: string;
  placeholder?: string;
  onChange?: (event: any) => void;
  style?: CSSProperties;
  icon?: string;
  type?: string;
}
