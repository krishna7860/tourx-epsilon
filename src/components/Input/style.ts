import styled from "styled-components";

export interface InputProps {
  ["data-icon"]: string | undefined;
}

export default styled.input`
  padding: 24px;
  padding-left: 64px;
  width: 100%;
  border: 1px solid #f2f4f8;
  color: black;
  outline: none;
  border-radius: 10px;
  ${(props: InputProps) =>
    props["data-icon"] &&
    `background: url("${props["data-icon"]}")
    no-repeat scroll 17px 20px;`}

  background-size: 28px;
  background-color: #f2f4f8;
  ::placeholder {
    color: #cfd0d7;
    font-size: 14px;
  }

  :focus {
    transition: all 0.1s;
    border: 2px solid #d5d8dc;
  }
`;
