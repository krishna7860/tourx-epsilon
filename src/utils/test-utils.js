/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { renderHook as rtlRenderHooks } from "@testing-library/react-hooks";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import * as themes from "../theme/schema.json";
import reducer from "../redux/rootReducer";
// import defaultStore from "../redux/defaultStore";

function render(ui, options = {}) {
  const {
    initialState = {},
    store = createStore(reducer, initialState),
    theme,
    ...renderOptions
  } = options;

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <ThemeProvider theme={themes.data.light}>{children}</ThemeProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// function renderHookWithRedux(callback, options = {}) {
//   const { initialState = defaultStore, store = createStore(reducer, initialState), ...others } = options

//   const returned = rtlRenderHooks(callback, {
//     ...others,
//     // eslint-disable-next-line react/display-name
//     wrapper: ({ children }) => <Provider store={store}>{children}</Provider>
//   })

//   return {
//     ...returned,
//     store
//   }
// }

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, rtlRender as renderHook };
