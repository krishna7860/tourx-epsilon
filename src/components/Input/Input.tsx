import React from "react";
import { InputProps } from "./Input.interface";
import StyledInput from "./style";

const Input = (props: InputProps): JSX.Element => {
  const { placeholder, onChange, style, icon, type } = props;

  return (
    <StyledInput
      placeholder={placeholder ?? ""}
      onChange={onChange}
      style={style}
      data-icon={icon}
      type={type}
      autoFocus
    />
  );
};

export default Input;
