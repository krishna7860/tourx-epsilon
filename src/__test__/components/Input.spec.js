/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { render } from "../../utils/test-utils";
import Input from "../../components/Input/Input";

describe("components/Input", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
