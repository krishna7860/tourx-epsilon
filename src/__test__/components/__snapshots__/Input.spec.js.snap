// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input should be rendered properly 1`] = `
<DocumentFragment>
  <input
    class="sc-bdnxRM iaTSye"
    placeholder=""
  />
</DocumentFragment>
`;
